src/SnapshotDB.php

```diff
@@ -97,6 +97,72 @@ public function insert( $table, $data ) {
 		return self::$dbo->lastInsertRowid();
 	}
 
+	/**
+	 * Generic method to insert details into the requested table.
+	 *
+	 * @param array $data Column data.
+	 *
+	 * @return mixed
+	 */
+	public function update_service_configuration( $data ) {
+		$table = 'snapshot_storage_credentials';
+
+		$is_aws_config_exist = $this->select(
```

Comment:
```suggestion
		$does_service_config_exist = $this->select(
```
Could we rename so it can be used by other services as well?